sdk/core/src/sleep.rs

```diff
@@ -0,0 +1,35 @@
+use futures::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::thread;
+use std::time::Duration;
+
+pub(crate) fn sleep(duration: Duration) -> Sleep {
+    Sleep {
+        thread: None,
+        duration,
+    }
+}
+
+pub(crate) struct Sleep {
+    thread: Option<thread::JoinHandle<()>>,
+    duration: Duration,
+}
+
+impl Future for Sleep {
```

Comment:
Is implementing our own Sleep future the right way to go? This is fairly expensive (one thread per sleep). But perhaps this is the best way to go about this....